src/routes/transactions/mappers/transactions-history.mapper.ts

```diff
@@ -188,27 +188,21 @@ export class TransactionsHistoryMapper {
     onlyTrusted: boolean,
   ): Promise<TransactionItem[]> {
     const limitedTransfers = transfers.slice(0, this.maxNestedTransfers);
-    const result: TransactionItem[] = [];
 
-    for (const transfer of limitedTransfers) {
-      const nestedTransaction = await this.transferMapper.mapTransfer(
-        chainId,
-        transfer,
-        safe,
-      );
-
-      const transferWithValue = this.mapZeroValueTransfer(nestedTransaction);
-      // If we do not have a transfer with value, we do not add it to the result
-      if (!transferWithValue) continue;
+    const nestedTransactions = await Promise.all(
+      limitedTransfers.map((transfer) =>
+        this.transferMapper.mapTransfer(chainId, transfer, safe),
+      ),
+    );
 
-      const trustedTransfer = onlyTrusted
-        ? this.mapTrustedTransfer(transferWithValue)
-        : transferWithValue;
+    return nestedTransactions
+      .filter((nestedTransaction) => {
+        // If we do not have a transfer with value, we do not add it to the result
+        if (!this.mapZeroValueTransfer(nestedTransaction)) return false;
 
-      if (!trustedTransfer) continue;
-      result.push(new TransactionItem(nestedTransaction));
-    }
-    return result;
+        return !onlyTrusted || this.mapTrustedTransfer(nestedTransaction);
```

Comment:
I find this line here a but confusing: the `filter` expects a predicate to filter the elements present in the collection.

However we are using the _truthy_ value with `this.mapTrustedTransfer(nestedTransaction)` so we might end with the following filter results:

- `true` – if we want to include all transactions.
- `false || Transaction` – if we want to include only trusted transactions – this results in `Transaction` being returned in the predicate.
- `false || null` – if we want to include only trusted transactions – this results in `null` being returned from the predicate.

I'd therefore suggest `isX` where `isX` returns `boolean` instead of `mapX` since using a `map` inside of a `filter` hints to some side-effects.

Reply:
> I'd therefore suggest `isX` where `isX` returns `boolean` instead of `mapX` since using a `map` inside of a `filter` hints to some side-effects.

Changed accordingly in d6e0feb49a00dafc31d09bb040f2f1bc883fe066.